### PR TITLE
DS-1394

### PR DESCRIPTION
**Pull request by thaywardIDBS**

### CODE REVIEW

File: reportportal_client/service.py [hunk 1]
```diff
@@ -14,198 +14,86 @@
...
         self.endpoint = endpoint
+        self.log_batch_size = log_batch_size
+        super(ReportPortalResultsReportingService, self).__init__(endpoint, token, verify_ssl=verify_ssl,
+                                                                  retries=retries)
         self.project = project
-        self.token = token
         self.is_skipped_an_issue = is_skipped_an_issue
         self.base_url_v1 = uri_join(self.endpoint, "api/v1", self.project)
         self.base_url_v2 = uri_join(self.endpoint, "api/v2", self.project)
+        self.base_project_url_v1 = self.base_url_v1
```
Comment (fheineIDBS) [inline reportportal_client/service.py]: Why two variables with the same value?

File: reportportal_client/service.py [hunk 2]
```diff
@@ -14,198 +14,86 @@
...
+                max_retries=retries, pool_maxsize=self.max_pool_size))
         self.session.headers["Authorization"] = "bearer {0}".format(self.token)
         self.launch_id = None
         self.verify_ssl = verify_ssl
 
+
     def terminate(self, *args, **kwargs):
         """Call this to terminate the service."""
         pass
+        self.base_project_url_v1 = uri_join(self.base_url_v1, self.project)
```
Comment (fheineIDBS) [inline reportportal_client/service.py]: Why is this here?

File: reportportal_client/service.py [hunk 3]
```diff
@@ -216,27 +104,183 @@ def start_launch(self,
...
+            "filter.eq.parentId": suite_id,
+            "filter.eq.type": "SUITE"
+        }
+        r = self.get_from_url(url, parameters=parameters)
+        content = _get_json(r)['content']
+
+        return content
+
+
+    def find_parent_suite_for_test(self, test_suite_path):
```
Comment (fheineIDBS) [inline reportportal_client/service.py]: Is this not a Robot Frameowrk thing? It so, it should be in the listener.

File: reportportal_client/service.py [hunk 4]
```diff
@@ -216,27 +104,183 @@ def start_launch(self,
...
+        root_suite = suite_path[0]
+
+        # Look in root for highest level suite
+        root_suites = self.get_root_suites()
+        current_suite_id = None
+        current_suite_uuid = None
+        found = False
+
+        # Look for top level suite in RP launch root suites. (suite is a dict)
+        for suite in root_suites:
```
Comment (fheineIDBS) [inline reportportal_client/service.py]: This code is duplicated here and blow for child suites. It should be extracted into another method. The logic that you have used here to create all the child suites if a suite isn't found also applies lower down the tree.

File: reportportal_client/service.py [hunk 5]
```diff
@@ -216,27 +104,183 @@ def start_launch(self,
...
+            # remove .robot from suite name
+            suite_name = suite['name']
+
+            if suite_name == root_suite:
```
Comment (fheineIDBS) [inline reportportal_client/service.py]: This is a case-sensitive comparison, the one for children is non-case sensitive. It should be one way or the other, but consistent.

File: reportportal_client/utilities.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+def _get_msg(response):
+    """
+    Get message from Response.
+
+    :param response: Response object
+    :return: data: json data
+    """
+    try:
+        return _get_data(response)
+    except KeyError:
```
Comment (fheineIDBS) [inline reportportal_client/utilities.py]: Where would this error come from? I can't see where that might get raised.

File: reportportal_client/utilities.py [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+                    slashes (will be stripped up).
+
+    Returns:
+        An uri string.
+
+    """
+    return '/'.join(str(s).strip('/').strip('\\') for s in uri_parts)
+
+
+def now():
```
Comment (fheineIDBS) [inline reportportal_client/utilities.py]: This is timezone sensitive and seems like a weird thing to do.

**Review by fheineIDBS — CHANGES_REQUESTED** [review]
Can you please also add at least some pyton unit tests? There is no need to go overbaord on those, but some baic stuff happy and error case stuff would be good.

File: reportportal_client/service.py (hunk 1 above)
Comment (thaywardIDBS) [reply reportportal_client/service.py]: Removed `self.project` from base url and added to base project urls

File: reportportal_client/service.py (hunk 2 above)
Comment (thaywardIDBS) [reply reportportal_client/service.py]: Removed, as it's not clear why the values of these variables needed to change here

File: reportportal_client/service.py (hunk 4 above)
Comment (thaywardIDBS) [reply reportportal_client/service.py]: new method created and used here and later in this method.

File: reportportal_client/utilities.py (hunk 6 above)
Comment (thaywardIDBS) [reply reportportal_client/utilities.py]: This is dealt with by extracting duplicated code to create new method.

File: tests/reportportal_clientTest.py [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+        test_path = [self.test_suite_1_name, self.test_suite_2_name, self.test_test_name]
+        parent_uuid = self.service.get_parent_suite_uuid(test_path)
+        assert parent_uuid == self.test_suite_2_uuid
+
+    def test_get_parent_suite_uuid_incorrect_path(self):
+        try:
+            test_path = ['foo', 'bar', 'test']
+            self.service.get_parent_suite_uuid(test_path)
+            error_message = None
+        except Exception as e:
```
Comment (fheineIDBS) [inline tests/reportportal_clientTest.py]: assertRaisesRegexp woud be a better way of doing this.

File: tests/reportportal_clientTest.py [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+    def test_start_test_item_test(self):
+        test_id = self.service.start_test_item('Start Test test', datetime.utcnow(), 'TEST', parent_item_id=self.test_suite_1_uuid)
+        assert UUID(test_id)
+
+    def test_start_test_item_test_invalid_parent(self):
+        try:
+            self.service.start_test_item('Start Test test', datetime.utcnow(), 'TEST',
+                                                   parent_item_id='52a568ec-fd8c-11ea-adc1-0242ac120002')
+            error_message = None
+        except Exception as e:
```
Comment (fheineIDBS) [inline tests/reportportal_clientTest.py]: assertRaisesRegexp would be a better way of doing this.

**Review by fheineIDBS — CHANGES_REQUESTED** [review]

**Review by fheineIDBS — APPROVED** [review]
Looks good now.
